src/fides/api/api/v1/endpoints/system.py

```diff
@@ -178,6 +178,12 @@ def patch_connection_secrets(
     for key, value in validated_secrets.items():
         connection_config.secrets[key] = value  # type: ignore
 
+    # Deauthorize an OAuth connection when the secrets are updated. This is necessary because
+    # the existing access tokens may not be valid anymore. This only applies to SaaS connection
+    # configurations that use the "oauth2_authorization_code" authentication strategy.
+    if connection_config.authorized:
```

Comment:
```suggestion
    if connection_config.authorized and connection_config.type is CONNECTION_TYPE.SAAS:
```

Does this update make sense? It'll make it explicit in the code that this is only supposed to happen to saas connectors

Reply:
Good callout! The more guardrails the better